router/gin/endpoint.go

```diff
@@ -109,8 +109,13 @@ func NewRequest(headersToSend []string) func(*gin.Context, []string) *proxy.Requ
 				headers[k] = h
 			}
 		}
+
 		headers["X-Forwarded-For"] = []string{c.ClientIP()}
-		headers["User-Agent"] = router.UserAgentHeaderValue
+		// if User-Agent is not forwarded using headersToSend, we set
+		// the KrakenD router User Agent value
+		if headers["User-Agent"] == nil {
+			headers["User-Agent"] = router.UserAgentHeaderValue
+		}
```

Comment:
also, I think we should let the backends know the request has been forwarded by the krakend, so I'd add an else statement
```
} else {
	headers["X-Forwarded-Via"] = router.UserAgentHeaderValue
}
```

Reply:
Would it makes sense to always add it (not only in the `else`)?

Reply:
If the `User-Agent` already contains the `UserAgentHeaderValue` I think we can't avoid the duplication